velox/expression/Expr.cpp

```diff
@@ -344,6 +344,14 @@ bool isFlat(const BaseVector& vector) {
       encoding == VectorEncoding::Simple::CONSTANT);
 }
 
+inline void checkResultInternalState(VectorPtr& result) {
+#ifndef NDEBUG
```

Comment:
nit: Can we define DEBUG instead of NDEBUG - find it harder to read double negative ..

Reply:
NDEBUG is standardized in cpp but DEBUG is not, see discussion on https://stackoverflow.com/questions/2290509/debug-vs-ndebug